python/cuml/cuml/linear_model/logistic_regression.pyx

```diff
@@ -301,12 +301,17 @@ class LogisticRegression(UniversalBase,
         # Converting y to device array here to use `unique` function
         # since calling input_to_cuml_array again in QN has no cost
         # Not needed to check dtype since qn class checks it already
-        y_m, n_rows, _, _ = input_to_cuml_array(y)
-        self.classes_ = cp.unique(y_m)
-        self._num_classes = len(self.classes_)
+        if isinstance(y, np.ndarray) and y.dtype.kind in 'SU':
+            self.classes__, y = np.unique(y, return_inverse=True)
+            self.numeric_classes_ = np.arange(len(self.classes__))
```

Comment:
What is the motivation for introducing new estimator attributes? I'd suggest that we just convert the provided labels into numeric classes and drop them otherwise.

Reply:
The motivation was to reproduce the behavior of a Scikit-Learn estimator :
```
>>> print(cuLogRegModel.classes_)
['setosa' 'versicolor' 'virginica']
```

But, to be frank this is a small detail and making it work involves solving a lot of edge cases in CI. Since we need this to be merged very soon, it is safer to go with a simpler version that just converts the inputs to numeric classes.

Reply:
I'm not sure that it is such a small detail, but I'd support merging the simpler version and leaving this as a follow-up.

Reply:
Why not use the `LabelEncoder` to handle all this complexity? It comes with the right value in its `classes_` attribute "for free".

If we merge something that is different from what scikit-learn does we just create more problems for ourselves IMHO. We put in effort to a half baked fix, we have to deal with users who report breakage and then we need to do the "right thing" anyway later.

Reply:
@betatim I'm in favor of using the LabelEncoder approach which was also previously suggested by @jcrist . @jcrist Do you already have an implementation that we can reference?

Reply:
The difficulty isn't in producing the unique classes, but to make sure that the numeric version of classes can be used internally by cuML while the text version can be displayed through the `classes_` attribute. Just pushed a solution that might pass CI. Then, sure we can use the `LabelEncoder` estimator instead of the unique function if that helps, it could especially handle https://github.com/rapidsai/cuml-accel/issues/94.

Reply:
I don't have an implementation yet, was going to finish up my other issue first. Haven't done more than look at how scikit-learn does it (with a `LabelEncoder`). I am wary of the additional fitted attribute here (`numeric_classes_` and `classes__`). The scikit-learn implementation doesn't need this extra state and manages to handle things just fine - it would be nice to avoid increasing the amount of state stored on the class if possible (but again, apologies, I haven't looked more into _how_ we might achieve that).